tests/test_metrics.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+from typing import Any
+
+from opentelemetry.metrics import CallbackOptions, UpDownCounter
+
+from appsignal.metrics import _counters, _gauges, _meter, increment_counter, set_gauge
+
+
+def test_increment_counter_creates_new_counter():
+    assert _counters.get("metric_name") is None
+
+    increment_counter("metric_name", 1)
+
+    assert isinstance(_counters["metric_name"], UpDownCounter)
+
+
+def test_increment_counter_updates_existing_counter(mocker):
+    increment_counter("metric_name", 1)
+
+    counter_spy = mocker.spy(_counters["metric_name"], "add")
+    increment_counter("metric_name", 1)
+
+    counter_spy.assert_called_once_with(1, None)
+
+
+def test_increment_counter_with_tags(mocker):
+    increment_counter("metric_name", 1)
+
+    counter_spy = mocker.spy(_counters["metric_name"], "add")
+
+    increment_counter("metric_name", 1, {"tag1": "value1"})
+
+    counter_spy.assert_called_once_with(1, {"tag1": "value1"})
+
+
+def test_set_gauge_creates_new_gauge(mocker):
+    meter_spy = mocker.spy(_meter, "create_observable_gauge")
+    assert "metric_name" not in _gauges
+
+    set_gauge("metric_name", 10)
+
+    # Registers the gauges internally
+    assert _gauges["metric_name"][None] == 10
+
+    # Check if it was registered on the meter
+    meter_spy.assert_called_once_with("metric_name", callbacks=mocker.ANY)
+    callbacks = meter_spy.call_args.kwargs["callbacks"]
+    assert len(callbacks) == 1
+
+    # Mock the ObservableGauge's async task calling the callbacks
+    return_values = callbacks[0](CallbackOptions())
+    assert len(return_values) == 1
+    return_value = return_values[0]
+    assert return_value.value == 10
+    assert return_value.attributes is None
```

Comment:
I think it would be good to assert that, after the gauge callback has been called, the corresponding entry in the gauge value dictionary is empty (`_gauges["metric_name"] == {}`) and/or that calling the callback again results in no observations being sent.

(Same for the `with_tags` scenario)

Reply:
Good catch! Amended extra assertions